PayPal/src/main/java/com/braintreepayments/api/PayPalCheckoutRequest.java

```diff
@@ -0,0 +1,246 @@
+package com.braintreepayments.api;
+
+import android.text.TextUtils;
+
+import androidx.annotation.StringDef;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Represents the parameters that are needed to start the PayPal Checkout flow
+ */
+public class PayPalCheckoutRequest extends PayPalRequest {
+
+    @Retention(RetentionPolicy.SOURCE)
+    @StringDef({PayPalCheckoutRequest.INTENT_ORDER, PayPalCheckoutRequest.INTENT_SALE, PayPalCheckoutRequest.INTENT_AUTHORIZE})
+    @interface PayPalPaymentIntent {}
+
+    /**
+     * Payment intent to create an order
+     */
+    public static final String INTENT_ORDER = "order";
+
+    /**
+     * Payment intent for immediate payment
+     */
+    public static final String INTENT_SALE = "sale";
+
+    /**
+     * Payment intent to authorize a payment for capture later
+     */
+    public static final String INTENT_AUTHORIZE = "authorize";
+
+    @Retention(RetentionPolicy.SOURCE)
+    @StringDef({PayPalCheckoutRequest.USER_ACTION_DEFAULT, PayPalCheckoutRequest.USER_ACTION_COMMIT})
+    @interface PayPalPaymentUserAction {}
+
+    /**
+     * Shows the default call-to-action text on the PayPal Express Checkout page. This option indicates that a final
+     * confirmation will be shown on the merchant checkout site before the user's payment method is charged.
+     */
+    public static final String USER_ACTION_DEFAULT = "";
+
+    /**
+     * Shows a deterministic call-to-action. This option indicates to the user that their payment method will be charged
+     * when they click the call-to-action button on the PayPal Checkout page, and that no final confirmation page will
+     * be shown on the merchant's checkout page. This option works for both checkout and vault flows.
+     */
+    public static final String USER_ACTION_COMMIT = "commit";
+
+    private String intent = INTENT_AUTHORIZE;
+    private String userAction = USER_ACTION_DEFAULT;
+    private final String amount;
+    private String currencyCode;
+    private boolean requestBillingAgreement;
+    private boolean offerPayLater;
+
+    /**
+     * This amount may differ slightly from the transaction amount. The exact decline rules
+     * for mismatches between this client-side amount and the final amount in the Transaction
+     * are determined by the gateway.
```

Comment:
Nitpick - it might be more clear to move this below the `@param amount` section

Reply:
Reordered this comment 👍